docs/concepts/controllers.md

```diff
@@ -0,0 +1,111 @@
+
+The logic relevant to the controller manager like the creation of the controller manager and registering each of the controllers with the manager, is contained in [`controllers/manager.go`](/controllers/manager.go).
+
+## Etcd Controller
+
+The *etcd controller* is responsible for the reconciliation of the `Etcd` resource.
+It handles the provisioning and management of the etcd cluster. Different components that are required for the functioning of the cluster like `Leases`, `ConfigMap`s, and the `Statefulset` for the etcd cluster are all deployed and  managed by the *etcd controller*.
+
+While building the controller, an event filter is set such that the behaviour of the controller depends on the `gardener.cloud/operation: reconcile` *annotation*. This is controlled by the `--ignore-operation-annotation` CLI flag, which, if set to `false`, tells the controller to perform reconciliation only when this annotation is present. If the flag is set to `true`, the controller will trigger reconciliation anytime the `Etcd` spec, and thus `generation`, changes.  
+
+The reason this filter is present is because any disruption in the `Etcd` resource due to reconciliation (due to changes in the `Etcd` spec, for example) while workloads are being run would be disastrous.
+Hence, any user who wishes to avoid such disruptions, can choose to set the `--ignore-operation-annotation` CLI flag to `false`. An example of this is Gardener's [gardenlet](https://github.com/gardener/gardener/blob/master/docs/concepts/gardenlet.md), which reconciles of the `Etcd` resource only during a shoot cluster's [*maintenance window*](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_maintenance.md).
+
+The controller adds a finalizer to the `Etcd` resource in order to ensure that the `Etcd` instance does not get deleted while the system is still dependent on the existence of the `Etcd` resource.
+Only the *etcd controller* can delete a resource once it adds finalizers to it. This ensures that the proper deletion flow steps are followed while deleting the resource. When the *etcd controller* enters the deletion flow, components are deleted in the reverse order that they were deployed in.
+
+The *etcd controller* is essential to the functioning of the etcd cluster and etcd-druid, thus the minimum number of worker threads is 1 (default being 3).
+
+## Custodian Controller
+
+The *custodian controller* acts on the `Etcd` resource.
+The primary purpose of the *custodian controller* is to update the status of the `Etcd` resource.
+
+It watches for changes in the status of the `Statefulset`s associated with the `Etcd` resources.
+Even though the `Etcd` resource owns the `Statefulset`, it is not necessary that the *etcd controller* reconciles whenever there are changes in the statuses of the objects that the `Etcd` resource owns.
+
+<!--- udpateEtcdStatus,-->
```

Comment:
Comment no longer needed?

Reply:
Missed it! Will remove it.

Reply:
Addressed in [a210894](https://github.com/gardener/etcd-druid/pull/722/commits/a2108945df415036f1622186141bddf4c74288b1).